docker_test.go

```diff
@@ -271,6 +292,64 @@ func run(m *testing.M) int {
 		log.Fatalf("Could not connect to docker: %s", err)
 	}
 
+	// pulls an redis image, creates a container based on it and runs it
+	resourceRedis, err := pool.Run("redis", "alpine3.14", []string{"requirepass=secret"})
+	if err != nil {
+		log.Fatalf("Could not start resource: %s", err)
+	}
+	defer func() {
+		if err := pool.Purge(resourceRedis); err != nil {
+			log.Printf("Could not purge resource: %s \n", err)
+		}
+	}()
+	// exponential backoff-retry, because the application in the container might not be ready to accept connections yet
+	address = fmt.Sprintf("localhost:%s", resourceRedis.GetPort("6379/tcp"))
+	if err := pool.Retry(func() error {
+		redisClient = redis.NewClient(&redis.Options{
+			Addr:     address,
+			Password: "",
+			DB:       0,
+		})
+
+		pong, err := redisClient.Ping().Result()
+		fmt.Println(pong, err)
+		return err
+	}); err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
+	network, err := pool.Client.CreateNetwork(dc.CreateNetworkOptions{Name: "coolest_network_ever"})
+	fmt.Println(err)
+	z, _ := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "confluentinc/cp-zookeeper",
+		Tag:        "latest",
+		NetworkID:  network.ID,
+		Hostname:   "zookeeper",
+		PortBindings: map[dc.Port][]dc.PortBinding{
+		   "2181/tcp": {{HostIP: "zookeeper", HostPort: "2181/tcp"}},
+		},
+		Env: []string{"ZOOKEEPER_CLIENT_PORT=2181"},
+	 })
+	// Set Kafka: pulls an image, creates a container based on it and runs it
+	KAFKA_ZOOKEEPER_CONNECT:=fmt.Sprintf("KAFKA_ZOOKEEPER_CONNECT= zookeeper:%s", z.GetPort("2181/tcp"))
+	fmt.Println(KAFKA_ZOOKEEPER_CONNECT)
```

Comment:
This will be useful for debugging, so I suggest keeping it, but:
```suggestion
	log.Println("KAFKA_ZOOKEEPER_CONNECT:", KAFKA_ZOOKEEPER_CONNECT)
```